### DISCUSSION TITLE
BlobWriter has no mime type when i read a image from docx file

### DISCUSSION THREAD

**Discussion by millet0328** [gildas-lormeau/zip.js#484] [General]
In the brower, when i read a image from docx file (zip file)，BlobWriter has no mime type。is there any other way I can't notice?
``` html5
<div class="hstack p-2 gap-2 bg-light">
	<input id="files" type="file" class="form-control" accept=".docx"/>
</div>
```
```javascript
	const fileInput = document.querySelector("#files");

	async function renderDocx(file) {
		let reader = new zip.ZipReader(new zip.BlobReader(file));
		const entries = await reader.getEntries();
		let blobWriter = new zip.BlobWriter();
		let image = await entries.find((entry) => entry.filename === "word/media/image1.png").getData(blobWriter);
		console.log(image);
	}

	fileInput.addEventListener("change", ev => {
		renderDocx(fileInput.files[0]);
	});
```
**I think we can detect the mime type by the magic number in buffer, just like [file-type](https://github.com/sindresorhus/file-type)**

**Comment by gildas-lormeau** [comment]
zip.js provides a function [`getMimeType()`](https://gildas-lormeau.github.io/zip.js/api/functions/getMimeType.html). You can use it to pass the MIME type to the [`BlobWriter()`](https://gildas-lormeau.github.io/zip.js/api/classes/BlobWriter.html#constructor.new_BlobWriter) constructor.

**Reply by millet0328** [reply]
I understand, thanks for your response.
